keras_nlp/layers/modeling/transformer_decoder.py

```diff
@@ -108,14 +108,15 @@ def __init__(
         kernel_initializer="glorot_uniform",
         bias_initializer="zeros",
         normalize_first=False,
+        name=None,
```

Comment:
I think this is a place we are not in KerasNLP, but I would say let's not do this to avoid clutter. Core Keras Dense, for example, does not do this https://github.com/keras-team/keras/blob/v3.0.5/keras/layers/core/dense.py#L33-L57

And it's not just `name, also `trainable`, `dtype`, `autocast`. Would be a pain to replicate these in each layer. It would be great on the docs side to figure out how to add these to all layers across the site, but let's not clutter the code.

Reply:
If you have time, feel free to remove this from other layers where we are doing it!

Reply:
Thanks @mattdangerw. In this case, one thing we could do to increase consistency is to drop `name` – and other optional arguments passed explicitly to `super().__init__`, for that matter – from all the doctsrings and code as well as expand the definition of `**kwargs`. For example, changing `FNetEncoder` from this:

```python
@keras_nlp_export("keras_nlp.layers.FNetEncoder")
class FNetEncoder(keras.layers.Layer):
    """FNet encoder.

    [...]

    Args:
        [...]
        name: string. The name of the layer. Defaults to `None`.
        **kwargs: other keyword arguments.

    [...]
    """

    def __init__(
        self,
        intermediate_dim,
        dropout=0,
        activation="relu",
        layer_norm_epsilon=1e-5,
        kernel_initializer="glorot_uniform",
        bias_initializer="zeros",
        name=None,
        **kwargs
    ):
        super().__init__(name=name, **kwargs)
```

to this:

```python
@keras_nlp_export("keras_nlp.layers.FNetEncoder")
class FNetEncoder(keras.layers.Layer):
    """FNet encoder.

    [...]

    Args:
        [...]
        **kwargs: other keyword arguments passed to `keras.layers.Layer`, including `name`, `trainable`, 
            `dtype', `autocast` etc.

    [...]
    """

    def __init__(
        self,
        intermediate_dim,
        dropout=0,
        activation="relu",
        layer_norm_epsilon=1e-5,
        kernel_initializer="glorot_uniform",
        bias_initializer="zeros",
        **kwargs
    ):
        super().__init__(**kwargs)
```

That way the code will be tidier but the documentation will be clear on how users can leverage additional keywords.

What is your take?

Reply:
One nit. Maybe let's not even mention `autocast`, that's more of a power user feature.

```
**kwargs: other keyword arguments passed to `keras.layers.Layer`, including `name`, `trainable`, and `dtype`.
```